src/lib/steps/rollup.ts

```diff
@@ -77,7 +70,42 @@ export async function rollup(opts: RollupOptions): Promise<void> {
     file: opts.dest,
     format: opts.format,
     banner: '',
-    globals: globals,
+    globals: moduleId => {
+      let regMatch;
+      if (regMatch = /^\@angular\/platform-browser-dynamic(\/?.*)/.exec(moduleId)) {
+        return `ng.platformBrowserDynamic${regMatch[1]}`.replace("/", ".")
+      }
+
+      if (regMatch = /^\@angular\/platform-browser(\/?.*)/.exec(moduleId)) {
+        return `ng.platformBrowser${regMatch[1]}`.replace("/", ".")
+      }
+
+      if (regMatch = /^\@angular\/(.+)/.exec(moduleId)) {
+        return `ng.${regMatch[1]}`.replace("/", ".")
+      }
+
+      if (/^rxjs\/[^\/]+$/.test(moduleId)) {
+        return "Rx";
+      }
+
+      if (/^rxjs\/(add\/)?observable/.test(moduleId)) {
+        return "Rx.Observable";
+      }
+
+      if (/^rxjs\/scheduler/.test(moduleId)) {
+        return "Rx.Scheduler";
+      }
+
+      if (/^rxjs\/symbol/.test(moduleId)) {
+        return "Rx.Symbol";
+      }
+
+      if (/^rxjs\/(add\/)?operator/.test(moduleId)) {
+        return "Rx.Observable.prototype";
+      }
+
+      return namedExternals[moduleId] || ''; // leave it up to rollup to guess the global name
+    },
```

Comment:
See above. Extract to a function / named lambda and then write a unit test for that piece of functionality?

```ts
const umdModuleIdStrategy = (moduleId: string): string => { /* ... */ }

/* ... */

describe(`umdModuleIdStrategy`, () => {
  it(`should map rxjs/add/operator/<foo> to Rx.Observable.<foo>`, () => {
    /* ... */
  });
});
```

???

Reply:
Wdyt?